### PR TITLE
Feature/search by artist

### LINKED ISSUE #35: Chore: rename search helper
src/Components/Search/Search.js
line 29 rename search helper function that sets state on App

### LINKED ISSUE #40: If search is executed on 'Details' componenet for an album the details component will render the new album. If an artists search send the user back to home page to display search if results exist.
either new details or taken back to new cards

### PR DESCRIPTION
**Pull request by Max9545**
# Selector FE PR

## What does this merge do? Check all that apply.
- [x] Adds a new feature
- [x] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## Summary of changes

- Adds search by artists feature and renders home page full of album cards based on artist search performed

## Notes for reviewer (any known bugs, areas of interest, etc.)

- Needs testing
- More error handling within component may be needed

## Next Steps (Future Iterations)

- Add error handling to overall results whether album or artist.
- Add search by genre, year, label, type of media

### CODE REVIEW

File: src/Components/AlbumCardsDisplay/AlbumCardsDisplay.js [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
+  })
+  console.log(data)
+  if (loading) return 'Loading...'
+  if (error) return `Error! ${error.message}`  
+
+
+  const renderGivenAlbums = (list) => {
+    console.log(list)
+    return list.map(album => {
+      return <AlbumCard  album={album}/>
```
Comment (monshan) [inline src/Components/AlbumCardsDisplay/AlbumCardsDisplay.js]: Would reccomend getting rid of all these console.logs() before merging

File: src/Components/AlbumCard/AlbumCard.js [hunk 2]
```diff
@@ -1,57 +1,80 @@
...
+  //   console.log(list)
+  //  return list.map(genre => <p className='card_genre' data-cy='card_genre'>{genre}</p>)
+  // }
+
+  // const displayArtists = (list) => {
+  //   return list.map(name => <p>{ name }</p>)
+  //  }
+
+
+   const displayImage = (image) => {
```
Comment (monshan) [inline src/Components/AlbumCard/AlbumCard.js]: Just ensuring that this commented code is either removed or activated where necessary

File: src/Components/App/App.js [hunk 3]
```diff
@@ -61,30 +64,23 @@ function App() {
...
           <Route exact path="/">
-            <AlbumCardsDisplay
-              titles={ titles }
-              favorites={ favorites }
-              toggleFav={ toggleFav }
-              determineFav={ determineFav }
-              isFavorite={ isFavorite }
-            />
+            <section className="glass">
+              {/* {artist.length ? <AlbumCardsDisplay titles={ search } artist={artist}/> : <AlbumCardsDisplay  titles={ titles }/> } */}
```
Comment (monshan) [inline src/Components/App/App.js]: Remove this line

File: src/Components/App/App.js [hunk 4]
```diff
@@ -61,30 +64,23 @@ function App() {
...
         <Switch>
           <Route exact path="/">
-            <AlbumCardsDisplay
-              titles={ titles }
-              favorites={ favorites }
-              toggleFav={ toggleFav }
-              determineFav={ determineFav }
-              isFavorite={ isFavorite }
-            />
+            <section className="glass">
```
Comment (monshan) [inline src/Components/App/App.js:73]: I removed this 'section' element and put the glass class on the AlbumCardsDisplay component to streamline as its being applied to all instances on an old branch that won't be merged, would reccomend doing it here as well

File: src/Components/Details/Details.js [hunk 5]
```diff
@@ -19,6 +19,12 @@ const Details = ({
 }) => {
   
   const [isFav, setIsFav] = useState(isFavorite(title));
+console.log(title)
```
Comment (monshan) [inline src/Components/Details/Details.js]: Delete this console.log()

File: src/Components/FavoriteAlbums/FavoriteAlbums.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import '../AlbumCard/AlbumCard.css'
+import FavoriteCard from '../FavoriteCard/FavoriteCard';
+
+function FavoriteAlbumsDisplay ({ favorites }) {
+
+
+  const renderGivenAlbums = (list) => {
+    console.log(list)
```
Comment (monshan) [inline src/Components/FavoriteAlbums/FavoriteAlbums.js]: remove this console.log()

File: src/Components/FavoriteCard/FavoriteCard.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+  if (error) return `Error! ${error.message}`  
+
+
+  const displayGenres = (list) => {
+   return list.map(genre => <p className='card_genre' data-cy='card_genre'>{genre}</p>)
+  }
+
+  const deconstruct = ( album ) => {
+    console.log(album.album)
```
Comment (monshan) [inline src/Components/FavoriteCard/FavoriteCard.js]: remove console.log()

File: src/Components/Search/Search.js [hunk 8]
```diff
@@ -1,14 +1,40 @@
...
+      event.preventDefault()
+      setSearchArtist(userQuery)
+      history.push(`/`) 
+    } 
+  
+    if (type === 'album') {
+      event.preventDefault()
+      history.push(`/${userQuery}`)
+      // setAlbumSearch(userQuery)
+      console.log('album')
```
Comment (monshan) [inline src/Components/Search/Search.js]: remove these of not being used

File: src/Components/Search/Search.js [hunk 9]
```diff
@@ -22,11 +48,15 @@ const Search = ({ setTitles }) => {
         list="albumTitles"
         onChange={(e) => setUserQuery(e.target.value)}
         />
-      <button
-        onClick={(e) => helpSetSearch(e)}
-        data-cy='search-submit'>
-          Search
-      </button>
+        <datalist id="albumTitles">
+          {/* { determineOptionStatus() } */}
```
Comment (monshan) [inline src/Components/Search/Search.js]: remove

**Review by monshan — CHANGES_REQUESTED** [review]
Nice work getting this new feature working despite having to split the component architecture, requesting the following changes:

- Remove noted console.log() and commented code
- Free to remove the section class='glass' and replace the 'glass' class onto the AlbunCardContainer

File: src/Components/App/App.js (hunk 3 above)
Comment (Max9545) [reply src/Components/App/App.js]: Done

File: src/Components/App/App.js (hunk 4 above)
Comment (Max9545) [reply src/Components/App/App.js:73]: Done

File: src/Components/Details/Details.js (hunk 5 above)
Comment (Max9545) [reply src/Components/Details/Details.js]: Done

File: src/Components/FavoriteCard/FavoriteCard.js (hunk 7 above)
Comment (Max9545) [reply src/Components/FavoriteCard/FavoriteCard.js]: Done

File: src/Components/Search/Search.js (hunk 8 above)
Comment (Max9545) [reply src/Components/Search/Search.js]: done

File: src/Components/Search/Search.js (hunk 9 above)
Comment (Max9545) [reply src/Components/Search/Search.js]: done

**Pull request merged**